third_party/proton/CMakeLists.txt

```diff
@@ -22,5 +22,10 @@ else()
   include_directories(${Python3_INCLUDE_DIRS})
 endif()
 
+# Check if the platform is MacOS
+if(APPLE)
+  set(PROTON_PYTHON_LDFLAGS "-undefined dynamic_lookup -flto")
```

Comment:
Do we really need LTO on macos?  That's just an optimization, and it will make builds slower, right?

Reply:
I grabbed it from triton's python linking code:

https://github.com/openai/triton/blob/b7555471025f9add67bf2f8a3373bda2e4fc6e5b/CMakeLists.txt#L258C52-L258C56

We probably won't execute code on mac os arm64 other than the interpreter.

Reply:
Yeah, I think lto is unnecessary.

> We probably won't execute code on mac os arm64 other than the interpreter.

Yeah, exactly, but we do build it all the time, so we shouldn't need lto.

Reply:
> it will make builds slower, right?

Ah, I may see the confusion.  What I mean is, it will take longer to build the program.  Then the program will run faster.

Reply:
Yes

Reply:
The last commit that changed it was three years ago

https://github.com/openai/triton/commit/8eb63bcb0189845f34d4db4cfcb08dc1bedde04b

Reply:
Can you please help me test it out on mac to see if removing lto has any side effects? I just returned the macbook to my friend...

Reply:
OK, why don't you check it in with LTO and then I'll remove it if it's a significant impact on build times.

Reply:
On my GH200

With lto:
```
real    1m14.819s
user    25m35.060s
sys     0m36.236s
```

Without lto:
```
real    1m14.450s
user    25m32.347s
sys     0m36.457s
```

Reply:
I confirmed that in a link-only build it also doesn't make a difference on my mac.  (I did this by modifying lib/Dialect/TritonGPU/IR/Dialect.cpp and building with `ninja` twice.  I have ccache enabled, so the second time it only links.  With and without lto, the second build took about 6.6s.)